spec/lib/auth0/mixins/validation_spec.rb

```diff
@@ -0,0 +1,207 @@
+      instance = Auth0::Algorithm::HS256.secret('secret')
+
+      expect(instance).to respond_to(:secret)
+    end
+
+    it 'is expected to return the secret' do
+      instance = Auth0::Algorithm::HS256.secret('secret')
+
+      expect(instance.secret).to eq('secret')
+    end
+  end
+
+  context 'parameters' do
+    it 'is expected to raise an error with a nil secret' do
+      expect { Auth0::Algorithm::RS256.jwks_url(nil) }.to raise_exception
+    end
+
+    it 'is expected to raise an error with an empty secret' do
+      expect { Auth0::Algorithm::RS256.jwks_url('') }.to raise_exception
+    end
+  end
+end
+
+describe Auth0::Algorithm::RS256 do
+  context 'class' do
+    it 'is expected to respond to :jwks_url' do
+      expect(Auth0::Algorithm::RS256).to respond_to(:jwks_url)
+    end
+
+    it 'is expected not to respond to :new' do
+      expect(Auth0::Algorithm::RS256).not_to respond_to(:new)
+    end
+  end
+
+  context 'instance' do
+    it 'is expected to respond to :jwks' do
+      instance = Auth0::Algorithm::RS256.jwks_url('jwks url')
+
+      expect(instance).to respond_to(:jwks)
+    end
+
+    it 'is expected to return a jwks' do
+      stub_jwks
+      instance = Auth0::Algorithm::RS256.jwks_url(JWKS_URL)
+
+      expect(instance.jwks).to have_key('keys') and contain_exactly(a_hash_including(kid: 'test-key-1'))
+    end
+  end
+
+  context 'parameters' do
+    it 'is expected to raise an error with a nil jwks_url' do
+      expect { Auth0::Algorithm::RS256.jwks_url(nil) }.to raise_exception
+    end
+
+    it 'is expected to raise an error with an empty jwks_url' do
+      expect { Auth0::Algorithm::RS256.jwks_url('') }.to raise_exception
+    end
+
+    it 'is expected to raise an error with a non integer lifetime' do
+      expect { Auth0::Algorithm::RS256.jwks_url('jwks url', '1') }.to raise_exception
+    end
+
+    it 'is expected to raise an error with a negative lifetime' do
+      expect { Auth0::Algorithm::RS256.jwks_url('jwks url', -1) }.to raise_exception
+    end
+  end
+
+  context 'cache' do
+    it 'is expected to fetch the jwks from the web when the cache is empty' do
+      stub_jwks
+      Auth0::Algorithm::RS256.jwks_url(JWKS_URL).jwks
+
+      WebMock.after_request do
+        expect(a_request(:get, JWKS_URL)).to have_been_made.once
+      end
+    end
+
+    it 'is expected to fetch the jwks from the web when the cache is expired' do
+      stub_jwks
+      instance = Auth0::Algorithm::RS256.jwks_url(JWKS_URL, lifetime: 0)
+      instance.jwks
+      instance.jwks
+
+      WebMock.after_request do
+        expect(a_request(:get, JWKS_URL)).to have_been_made.twice
+      end
+    end
+
+    it 'is not expected to fetch the jwks from the web when there is a value cached' do
+      stub_jwks
+      instance = Auth0::Algorithm::RS256.jwks_url(JWKS_URL)
+      instance.jwks
+      instance.jwks
+
+      WebMock.after_request do
+        expect(a_request(:get, JWKS_URL)).to have_been_made.at_most_once
```

Comment:
let's be more explicit here, only one request should come through 

```suggestion
        expect(a_request(:get, JWKS_URL)).to have_been_made.once
```